### PR TITLE
Add ring buffer tests

### PR DESCRIPTION
**Pull request by gtrevi**
This PR adds two more tests to the ring buffer (for both Linux and Windows), running concurrently on all CPUs and averaging buffer sizes of ~150/200Mb:

- 300K concurrent iterations effectively writing 400 bytes.
- 100K concurrent iterations effectively writing 1420 bytes (i.e. a similar MTU size).

Each test will run with a bpf program that will appropriately size the ring buffer so to actually write *all* the payloads for each and every iteration, therefore without any skipping due the ring buffer being filled.

This is useful to emulate scenarios where multiple concurrent producers of data (e.g., packets) of different sizes need to store them in a ring-buffer map, and for which understanding the ring buffer latency is important.

Additionally, the PR adds flexible parameter number support, for tests requiring more than one parameter.

### CODE REVIEW

**Comment by Alan-Jowett** [comment]
Please rebase or merge.

**Comment by Alan-Jowett** [comment]
@gtrevi Please rebase or merge.

File: bpf/CMakeLists.txt [hunk 1]
```diff
@@ -5,61 +5,59 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
...
-    "baseline"
+    "baseline,baseline,-DBPF"
     "generic_map,hash,-DTYPE=BPF_MAP_TYPE_HASH"
     "generic_map,percpu_hash,-DTYPE=BPF_MAP_TYPE_PERCPU_HASH"
     "generic_map,lru_hash,-DTYPE=BPF_MAP_TYPE_LRU_HASH"
     "generic_map,lru_per_cpu_hash,-DTYPE=BPF_MAP_TYPE_LRU_PERCPU_HASH"
     "generic_map,array,-DTYPE=BPF_MAP_TYPE_ARRAY"
     "generic_map,percpu_array,-DTYPE=BPF_MAP_TYPE_PERCPU_ARRAY"
-    "helpers"
+    "helpers,helpers,-DBPF"
```
Comment (Alan-Jowett) [inline bpf/CMakeLists.txt:17]: What's with the extra -DBPF here?

Comment (gtrevi) [reply bpf/CMakeLists.txt:17]: In the previous implementation, `-DBPF` was added to all entries with one element, thus, porting the same behavior on the current one.

**Review by Alan-Jowett — APPROVED** [review]

**Pull request merged**